docs/developer-guide/etendo-classic/how-to-guides/how-to-create-junit-testcases.md

```diff
@@ -0,0 +1,525 @@
+
+  * a transaction is automatically started at first database access in the test cases. This is done by the Data Access Layer. 
+  * a transaction is either committed (when no exception happened) or rolled-back (when an exception happened). 
+
+The WeldBaseTest class detects automatically if an exception happened or not.
+
+There are certainly cases whereby it makes sense to have more control over the
+database transactions. There are a number of relevant methods which can be
+useful then:
+
+  * **OBDal.getInstance().flush()**: flushes the update/insert queries in hibernate to the database. 
+  * **OBDal.getInstance().commitAndClose()**: commits the transaction and closes the session. A new session/transaction is automatically started at the next database access. 
+  * **OBDal.getInstance().rollbackAndClose()**: rolls back and closes the transactions. A new session/transaction is automatically started at the next database access. 
+
+!!! info
+    DAL event observers  are not triggered within test cases extending ` OBBaseTest ` class. They to work require of test cases extending ` WeldBaseTest `. 
+
+###  Side-Effect Free
+
+A test case will often change the data in the underlying database. Most of the
+time it is not feasible to setup a completely new test database for each test
+run. Therefore test-cases should be developed such that they are side effect
+free. This means:
+
+  * When the test-case changes data then it should have a test method which is run as the last test method which cleans up/repairs the data. 
+  * This clean-up method should also clean up data which is left from previous test runs. For this common issue should be used _AfterClass_ notation. This method runs automatically at the end of the class.
+
+This last point is important because there can be always reasons why during a
+test the clean-up step is not performed. For example because the test run is
+stopped before the clean-up is done.
+
+###  Approach to features of JUnit
+
+####  Parameterized Tests
+
+!!! info
+    More info in:  [Parameterized-Test](https://github.com/junit-team/junit4/wiki/Parameterized-tests){target="_blank"}
+
+####  Rules
+
+!!! info
+    More info in:  [Rules](https://github.com/junit-team/junit4/wiki/Rules){target="_blank"}
+
+####  Assertions and Hamcrest 1.13
+
+Hamcrest is a framework for writing matcher objects allowing 'match' rules to
+be defined declaratively. There are a number of situations where matchers are
+invaluble, such as UI validation, or data filtering, but it is in the area of
+writing flexible tests that matchers are most commonly used.
+
+When writing tests it is sometimes difficult to get the balance right between
+overspecifying the test, and not specifying enough (making the test less
+valuable). Having a tool that allows you to pick out precisely the aspect
+under test and describe the values it should have, to a controlled level of
+precision, helps greatly in writing tests.
+
+!!! info
```

Comment:
@facumoyanosmf I would move this block to after the title instead of at the end.